### PR TITLE
migration to PyQt5

### PR DESCRIPTION
**Pull request by SomberNight**
This would greatly benefit from some testing. Basic functionality seems to work on my machine.

I could not always see a clean way of migrating old-style signals to new-style signals, hence the occasional ugliness.

At the very least, the following is missing:
- the Windows building stuff in `./contrib/build-wine/*`
- updating `./snap/snapcraft.yaml` (also for py3)

### CODE REVIEW

File: gui/qt/qrcodewidget.py [hunk 1]
```diff
@@ -109,12 +111,12 @@ def __init__(self, data, parent=None, title = "", show_text=False):
             filename = os.path.join(config.path, "qrcode.png")
 
             def print_qr():
-                p = QPixmap.grabWindow(qrw.winId())
+                p = QScreen.grabWindow(qrw.winId())
```
Comment (SomberNight) [inline gui/qt/qrcodewidget.py]: This does not work it seems. (Same for the one below) Needs a fix.
```
TypeError: grabWindow(self, sip.voidptr, x: int = 0, y: int = 0, width: int = -1, height: int = -1): first argument of unbound method must have type 'QScreen'
```

Comment (SomberNight) [inline gui/qt/qrcodewidget.py]: This does not work it seems. (Same for the one below) Needs a fix.
```
TypeError: grabWindow(self, sip.voidptr, x: int = 0, y: int = 0, width: int = -1, height: int = -1): first argument of unbound method must have type 'QScreen'
```

**Comment by SomberNight** [comment]
As can be seen from all the commits, I keep finding changes that need to be done.
I would be glad if someone else could also spend some time trying to get the GUI to crash (or produce errors).
In any case, I think this PR is more-or-less stable now.

File: gui/qt/qrcodewidget.py (hunk 1 above)
Comment (Wullee) [reply gui/qt/qrcodewidget.py]: @@ -109,12 +111,12 @@ def __init __ (ตัวเองข้อมูล parent = ไม่มี, title = "", show_text = เท็จ):
             filename = os.path.join (config.path, " qrcode.png " )
 
             def  print_qr ():
+ p =  QScreen .grabWindow (qrw.winId ())

**Pull request merged**